toxcore/group.h

```diff
@@ -28,104 +28,134 @@
 #include "Messenger.h"
 
 enum {
-    GROUPCHAT_STATUS_NONE,
-    GROUPCHAT_STATUS_VALID,
-    GROUPCHAT_STATUS_CONNECTED
-};
-
-enum {
     GROUPCHAT_TYPE_TEXT,
     GROUPCHAT_TYPE_AV
 };
 
 #define MAX_LOSSY_COUNT 256
 
 typedef struct {
+
+    uint8_t     recv_lossy[MAX_LOSSY_COUNT];
+    uint16_t    bottom_lossy_number, top_lossy_number;
+
+} Group_Peer_Lossy;
+
+typedef struct {
     uint8_t     real_pk[crypto_box_PUBLICKEYBYTES];
     uint8_t     temp_pk[crypto_box_PUBLICKEYBYTES];
 
+    Group_Peer_Lossy *lossy; /* rare use */
+    void *object;
+
     uint64_t    last_recv;
-    uint32_t    last_message_number;
 
-    uint8_t     nick[MAX_NAME_LENGTH];
-    uint8_t     nick_len;
+    uint8_t     *nick;
 
-    uint16_t peer_number;
+    uint32_t    last_message_number[9]; /* 9 - number of group messages */
+    int         friendcon_id;
 
-    uint8_t  recv_lossy[MAX_LOSSY_COUNT];
-    uint16_t bottom_lossy_number, top_lossy_number;
+    signed      gid : 24; /* unique per-conference peer id */
```

Comment:
gid is 16 bit unsigned, it is right. There are meaningful values [0..65535]. But I need addition one, -1, to indicate gid is not set. So 17 bits will be enough, but do not forget about bit alignments. You can believe: 24 bits - better.